docs/user-guide/model.rst

```diff
@@ -0,0 +1,173 @@
+##########
+Data Model
+##########
+
+Felis's `data model <../dev/internals.html#module-felis.datamodel>`__ is defined by a set of
+`Pydantic <https://docs.pydantic.dev/latest/>`__ classes which describe the semantics of tabular data.
+In addition to the standard conceptual constructs of a relational database schema, such as tables and columns,
+Felis provides a way to attach extra metadata like units of measurement on columns.
```

Comment:
... extra metadata to these elements, like units ...

(This avoids a grammatical implication that the "extra metadata" applies only to columns.)